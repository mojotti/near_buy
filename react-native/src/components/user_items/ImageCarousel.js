import React from 'react';
import PropTypes from 'prop-types';
import Carousel from 'react-native-looped-carousel';
import { styles } from '../../static/styles/UserItemDetailsStyles';
import DetailImage from './DetailImage';

export default class ImageCarousel extends React.Component {
  render() {
    return (
      <Carousel
        style={styles.carousel}
        autoplay={false}
        pageInfo
        currentPage={0}
        pageInfoBackgroundColor={'#292929'}
        pageInfoTextStyle={{ color: '#FFFFFF' }}
    // arrows are disabled for now since they don't work as intended on ios
        arrows={false}
        arrowStyle={styles.arrowStyle}
        leftArrowText={' '}
        rightArrowText={' '}
      >
        <DetailImage
          url={this.props.imageUrls[0]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[1]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[2]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[3]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
      </Carousel>
    );
  }
}

ImageCarousel.propTypes = {
  imageUrls: PropTypes.arrayOf(PropTypes.string).isRequired,
  id: PropTypes.number.isRequired,
  onImageUpload: PropTypes.func.isRequired,
  numOfPics: PropTypes.number.isRequired,
};
