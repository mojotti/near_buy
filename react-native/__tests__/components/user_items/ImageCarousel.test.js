import React from 'react';
import { shallow } from 'enzyme';
import ImageCarousel from '../../../src/components/user_items/ImageCarousel';


describe('ImageCarousel', () => {
  test('renders images', () => {
    const imageUrls = ['foo.com/png', 'bar.com/png'];
    const imageCarousel = shallow(
      <ImageCarousel
        imageUrls={imageUrls}
        id={0}
        onImageUpload={() => {}}
        numOfPics={2}
      />
    );

    expect(imageCarousel).toMatchSnapshot();
  });
});
