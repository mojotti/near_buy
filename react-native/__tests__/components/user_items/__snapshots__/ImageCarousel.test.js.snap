// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel renders images 1`] = `
<Carousel
  arrowStyle={
    Object {
      "backgroundColor": "transparent",
      "height": 667,
      "width": 225,
    }
  }
  arrows={false}
  autoplay={false}
  bullets={false}
  currentPage={0}
  delay={4000}
  isLooped={true}
  leftArrowText=" "
  pageInfo={true}
  pageInfoBackgroundColor="#292929"
  pageInfoTextSeparator=" / "
  pageInfoTextStyle={
    Object {
      "color": "#FFFFFF",
    }
  }
  rightArrowText=" "
  style={
    Object {
      "height": 667,
      "width": 750,
    }
  }
  swipe={true}
>
  <Connect(_DetailImage)
    id={0}
    numOfPics={2}
    onImageUpload={[Function]}
    url="foo.com/png"
  />
  <Connect(_DetailImage)
    id={0}
    numOfPics={2}
    onImageUpload={[Function]}
    url="bar.com/png"
  />
  <Connect(_DetailImage)
    id={0}
    numOfPics={2}
    onImageUpload={[Function]}
  />
  <Connect(_DetailImage)
    id={0}
    numOfPics={2}
    onImageUpload={[Function]}
  />
</Carousel>
`;
